connect/runtime/src/main/java/org/apache/kafka/connect/storage/KafkaConfigBackingStore.java

```diff
@@ -569,8 +559,7 @@ public void onCompletion(Throwable error, ConsumerRecord<String, byte[]> record)
 
                     // Validate the configs we're supposed to update to ensure we're getting a complete configuration
                     // update of all tasks that are expected based on the number of tasks in the commit message.
-                    Map<String, Set<Integer>> updatedConfigIdsByConnector = taskIdsByConnector(deferred);
-                    Set<Integer> taskIdSet = updatedConfigIdsByConnector.get(connectorName);
+                    Set<Integer> taskIdSet = taskIds(deferred);
```

Comment:
This can no longer return `null` based on the current implementation, so the subsequent check no longer makes sense. Although it looks like it was already impossible with the `taskIdsByConnector` code. I can't remember the exact context of KAFKA-3321 (I know we added this to help @gwenshap debug a transient system test failure). We might want to figure out if we can get rid of the check for a null `taskIdSet` now.